Lib/http/cookiejar.py

```diff
@@ -1044,12 +1044,13 @@ def set_ok_domain(self, cookie, request):
             else:
                 undotted_domain = domain
             embedded_dots = (undotted_domain.find(".") >= 0)
-            if not embedded_dots and domain != ".local":
+            if not embedded_dots and not erhn.endswith(".local"):
```

Comment:
Previously this checked a domain that came from the cookie, and now it checks the hostname from the request. Can you explain why this is changed and why the change is safe?

Reply:
This check considers `localhost` to be a non-local domain. It also doesn't have any embedded_dots, so it doesn't process it, which really shouldn't happen. My change should be safe because for local domains effective request hostname would end with ".local" (ERHN will end with .local if it ls clearly specified as part of the domain (`foo.local`), or if the domain consists of only one level (`localhost` has EHRN of `localhost.local`)).

Reply:
Thanks! I'm still worried about cases where the domain and the ERHN don't match (e.g. a malicious website `evil.com` sets a cookie for `localhost`). I see there are some checks for that and a test in `test_wrong_domain`, but it's not very thorough. Could you add more tests for similar cases (request to non-local domain setting a local cookie, and vice versa)?

Reply:
I've added some tests for those cases in [a31fa40](https://github.com/python/cpython/pull/30108/commits/a31fa402ebf805692cc991f04751067d66b46b38). Looks like everything is fine.